docs/user/advanced_interfaces.rst

```diff
@@ -0,0 +1,155 @@
+.. _advanced_interfaces:
+
+Advanced Interfaces
+===================
+
+Python and R are two of the most common programming languages for data science,
+and correspondingly JupyterLab and RStudio are common IDEs (Interactive Development
+Environments) for exploratory research. In Renku, we provide project `templates <templating>`_
+to easily set up one of these environments.
+
+But what if you want to use a different interface, like MATLAB? Or what if you
+want to install software that requires extra non-python or R dependencies on the
+machine?
+
+Dockerfile modifications
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In Renku we use Docker for containerization. While we
+have a set of defaults that we build into a minimal Python and R images
+image, there are several reasons why you might want to build on top of these or
+write your own entirely.
+
+Dockerfile structure
+""""""""""""""""""""
+
+The project's ``Dockerfile`` lives in the top level of the project directory. In the
+default ``Dockerfile`` provided in the template, the first line is a ``FROM``
+statement that specifies a `versioned base docker image <https://github.com/SwissDataScienceCenter/renku-jupyter>`_,
+that we do make updates to periodically. The heart of it is the set of installations of
+jupyterlab/rstudio, git, and renku::
+
+  FROM renku/singleuser:0.3.5-renku0.5.2
+
+  # or, for RStudio in the build
+
+  FROM renku/singleuser-r:0.3.5-renku0.5.2
+
+The next two statements install user-specified libraries from ``environment.yml``
+and ``requirements.txt``::
```

Comment:
```suggestion
and ``requirements.txt``:
```

Reply:
I double colon'd in a ton of places; I think sphinx parses these equivalently?

Reply:
Actually I think you need the double colon or it doesn't show up as a code block